packages/front-end/components/Settings/ApiKeys.tsx

```diff
@@ -60,6 +74,20 @@ const ApiKeys: FC = () => {
                   </Link>
                 </td>
                 <td>{key.description}</td>
+                <td>
+                  {key.encryptFeatures && !secretKeys[key.key] && (
+                    <button
+                      type="button"
+                      className="btn btn-link text-dark"
+                      onClick={() => revealSecret(key.key)}
+                    >
```

Comment:
If the API request is slow for whatever reason, there will be no indication to the user that it's working in the background.

You could use the `Button` component instead, which takes an async onClick handler and shows a loading state while it's running.

Reply:
Good call.